app/controllers/actions/GroupAction.scala

```diff
@@ -69,6 +70,18 @@ class GroupAction @Inject()
     }
   }
 
+  def withGroupNameAndAuthorised(body: (String, Arn) => Future[Result])
+                                          (implicit ec: ExecutionContext, hc: HeaderCarrier,
+                                           request: MessagesRequest[AnyContent], appConfig: AppConfig): Future[Result] = {
+    authAction.isAuthorisedAgent { arn =>
+      isOptedInWithSessionItem[String](GROUP_NAME)(arn) { maybeGroupName =>
+        maybeGroupName.fold(Redirect(controllers.routes.CreateGroupSelectNameController.showGroupName).toFuture) {
```

Comment:
I think the fold Redirect should be CreateGroupSelectGroupTypeController.showSelectGroupType - since showGroupName is custom only